jaeger-context/src/main/java/com/uber/jaeger/context/ActiveSpanSourceTraceContext.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2017, Uber Technologies, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.uber.jaeger.context;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.ActiveSpanSource;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+import io.opentracing.util.ThreadLocalActiveSpan;
+
+import java.lang.reflect.Field;
+
+/**
+ * This is a {@link TraceContext} that relies on the {@link ActiveSpanSource} registered with {@link
+ * GlobalTracer}.
+ */
+public class ActiveSpanSourceTraceContext implements TraceContext {
```

Comment:
Are you sure we need a new class instead of changing ThreadLocalTraceContext?  If someone is already using ThreadLocalTraceContext directly, then it won't work correctly any longer, so I think it warrants a breaking API change (namely having an argument to the constructor).

Reply:
We could also simply remove the `ThreadLocalTraceContext` instead of marking it `@Deprecated`

Reply:
why create extra churn with classes/filenames/history when we can reuse the same class and re-implement it as needed?

Reply:
Could you provide justification for sticking with the old name? 

The new name is more descriptive. 

Javac itself doesn't care about what this class used to be called. 

As far as git goes, it uses binary blobs to track file content. I expect it to be able to handle this operation efficiently. Have you seen examples where it hasn't? Should we be optimizing for git history size?